### PR TITLE
Allow specifying an appIdSuffix to run-android

### PR DESCRIPTION
**Pull request by Phredward**
Motivation

Currently react-native run-android instals and starts your app, but if you have more than one variant, it doesn't know which one to start.  This allows developers to pass in the suffix specified in build.gradle, so that the correct app is started.

Test Plan

verify that `react-native run-android` runs properly 
verify that `react-native run-android --appIdSuffix validSuffix` runs properly

### CODE REVIEW

**Comment by ericvicenti** [comment]
This is pretty well-documented and the test plan seems reasonable. Thanks for the contribution, I'll land it now!

**Comment by karanjthakkar** [comment]
@Phredward I think you would find this implementation more useful since it cuts down on adding a new option: https://github.com/facebook/react-native/pull/12820 It uses the package name defined in the generated manifest file instead of manually specifying the buildType, flavor, etc.

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -150,9 +152,9 @@ function tryInstallAppOnDevice(args, device) {
   }
 }
 
-function tryLaunchAppOnDevice(device, packageName, adbPath, mainActivity) {
+function tryLaunchAppOnDevice(device, packageNameWithSuffix, packageName, adbPath, mainActivity) {
   try {
-    const adbArgs = ['-s', device, 'shell', 'am', 'start', '-n', packageName + '/.' + mainActivity];
+    const adbArgs = ['-s', device, 'shell', 'am', 'start', '-n', packageNameWithSuffix + '/' + packageName +  + mainActivity];
```
Comment (karanjthakkar) [inline local-cli/runAndroid/runAndroid.js]: @Phredward This be `+ '.' + mainActivity` right? With the current implementation it breaks and searches for `≤packageName>NaN` activity. cc: @ericvicenti

Comment (Phredward) [inline local-cli/runAndroid/runAndroid.js]: Yup.  Thanks for catching that.  You can see that the other shell am start (with the hardcoded   '.MainActivity') gets it right.

Can you make this fix, or should I do it?

Comment (karanjthakkar) [inline local-cli/runAndroid/runAndroid.js]: @Phredward Yeah. I saw that. I won't be able to make any changes to this PR. If you are unable to do it, I can do it in a separate PR but I would rather the authorship stay with you for working on this :)

Comment (Phredward) [inline local-cli/runAndroid/runAndroid.js]: Done

**Comment by karanjthakkar** [comment]
@Phredward LGTM 👍 Thanks for this!

**Pull request closed without merging**